File: spec/java/src/io/kaitai/struct/spec/TestSwitchManualEnumInvalid.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.SwitchManualEnumInvalid;
import org.testng.annotations.Test;
import static org.testng.Assert.*;
public class TestSwitchManualEnumInvalid extends CommonSpec {
    @Test
    public void testSwitchManualEnumInvalid() throws Exception {
        SwitchManualEnumInvalid r = SwitchManualEnumInvalid.fromFile(SRC_DIR + "enum_negative.bin");
        assertIntEquals(r.opcodes().size(), 2);
        assertNull(r.opcodes().get(((int) 0)).code());
        assertNull(r.opcodes().get(((int) 0)).body());
        assertEquals(r.opcodes().get(((int) 1)).code(), SwitchManualEnumInvalid.Opcode.CodeEnum.FOO);
        assertNull(r.opcodes().get(((int) 1)).body());
    }
}

File: spec/python/spec/test_switch_manual_enum_invalid.py
import unittest
from testformats.switch_manual_enum_invalid import SwitchManualEnumInvalid
from testformats import switch_manual_enum_invalid

class TestSwitchManualEnumInvalid(unittest.TestCase):
    def test_switch_manual_enum_invalid(self):
        with SwitchManualEnumInvalid.from_file('src/enum_negative.bin') as r:
            self.assertEqual(len(r.opcodes), 2)
            self.assertEqual(r.opcodes[0].code, 255)
            self.assertFalse(hasattr(r.opcodes[0], 'body'))
            self.assertEqual(r.opcodes[1].code, switch_manual_enum_invalid.SwitchManualEnumInvalid.Opcode.CodeEnum.foo)
            self.assertFalse(hasattr(r.opcodes[1], 'body'))
